DXRando/DeusEx/Classes/Keypad.uc

```diff
@@ -33,8 +34,13 @@ function bool GetInstantSuccess(DeusExPlayer Hacker, bool bHacked)
 function RunEvents(DeusExPlayer Player, bool bSuccess)
 {
     super.RunEvents(Player,bSuccess);
-    if (bSuccess && !WasHacked()){
-        bCodeKnown = True;
+    if (bSuccess){
+        if ( !WasHacked() ) {
+            bCodeKnown = True;
+        }
+        if (bUnlock) {
```

Comment:
`bUnlock` is checked inside the function already so this is redundant

I would say either delete the check here and rename the function to `CheckUnlockDoor()` or something

or delete the check inside the function
```suggestion
```